### ISSUE TITLE
lansection with sigleside

### ISSUE THREAD

**Issue by QLwin** [esmini/esmini#366]
Whether to support the processing of different left and right section？ 

lansection with sigleside 

![image](https://user-images.githubusercontent.com/35674481/208609527-5fed14cc-09bf-4939-b952-ffc3a1f86498.png)

**Comment by eknabevcc** [comment]
This is a really useful feature of OpenDRIVE. Unfortunately not currently supported in esmini.
It's in the backlog to be implemented, but no estimated time plan.

**Comment by QLwin** [comment]
Is there a plan to implement this feature now? I find this is the only way to create a complete interchange.

**Comment by eknabevcc** [comment]
It's slightly moving up in backlog, but no date yet.

We haven't, so far, found it a blocker. Is it possible to explain the problem you encountered in a simple way? (it could influence our motivation)

**Comment by QLwin** [comment]
![image](https://github.com/esmini/esmini/assets/35674481/5dd16647-c553-4efc-ad7f-814fade4b3f3)
This is a highway with a central divider, and when the exit and entrance ramps are close to each other, it would be best to divide the section into left and right sides. The method I can think of now is to split Road01 into two roads, one for each side.

**Comment by QLwin** [comment]
![image](https://github.com/esmini/esmini/assets/35674481/3fe2f0ca-4d2d-4a2a-930d-365a5774f6f5)

I've drawn another scenario where there is a slight misalignment between the bus bays on the left and right sides of the road. Due to the addition of lanes, should we also consider the issue of sections?

**Comment by eknabevcc** [comment]
I agree the first example would be easier to implement using single side lane sections 👍
We'll keep this in mind for coming prioritizations. It seems like an useful feature.

Your suggestion seems like a good workaround for now, to specify each direction as separate roads. 

Regarding the bus bays, maybe this could be achieved in terms of laneWidth and roadMark elements...?

**Comment by QLwin** [comment]
thank you for your reply：）
If lanes are added or reduced on a section of the road, and their left and right stations are not aligned, the method of modifying laneWidth may not be applicable. Unfortunately, ASAM does not seem to provide specific cases when describing this function.

**Comment by eknabevcc** [comment]
Maybe a hack, but lanes don't need to be added or removed. They can be present all through the lane section, with width = 0 for sub-segments. Then just expand width by polynomial whenever needed. 

See road 202 in [multi_intersections.xodr](https://github.com/esmini/esmini/blob/master/resources/xodr/multi_intersections.xodr) as example. Lane 1 has width = 3.375m for first 33.5 meters (starting from intersection), then narrowing to width = 0.
```xml
<width sOffset="0.0000000000000000e+00" a="3.7500000000000000e+00" b="0.0000000000000000e+00" c="0.0000000000000000e+00" d="0.0000000000000000e+00"/>
<width sOffset="3.3500000000000000e+01" a="3.7500000000000000e+00" b="0.0000000000000000e+00" c="-1.7301038062283738e-02" d="4.5231472058258139e-04"/>
<width sOffset="5.9000000000000000e+01" a="0.0000000000000000e+00" b="0.0000000000000000e+00" c="0.0000000000000000e+00" d="0.0000000000000000e+00"/>
```

![Image](https://github.com/esmini/esmini/assets/29146839/12954b3d-56df-4154-ba88-0a0db8f01a9e)

**Comment by eknabevcc** [comment]
Good point. The type of lane will not change. So if any populated autonomous road users can detect zero lane widths and avoid those, then it may work, otherwise not I guess.

**Comment by csawp** [comment]
I have encountered the same issue. Currently, I’m solving it by dividing it into two separate lanes. If using lanes with a width of 0, there are two problems: firstly, an increase in OSI computation, and secondly, difficulty in accurately controlling the road markings.

**Comment by eknabevcc** [comment]
Regarding road marks: You can specify starting s-value for each roadmark segment. That way you can have various roadmark configurations (even empty) along a single lane section.
